### PR TITLE
Refactor images to build Varnish from source using the latest Alpine base image

### LINKED ISSUE #29: Refactor images to build Varnish from source using the latest Alpine base image

### PR DESCRIPTION
**Pull request by sergey-zabolotny**
Closes #29

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
-
-RUN apk add --update --no-cache \
-	bash \
-	varnish \
-	&& rm -rf /var/cache/apk/*
-
-COPY default.vcl /opt/default.vcl
-COPY startup.sh /opt/startup.sh
-COPY healthcheck.sh /opt/healthcheck.sh
+FROM alpine:3.8
```
Comment (lmakarov) [inline Dockerfile]: We should use the latest stable alpine version (3.9) unless there are major reasons not to.

File: .travis.yml [hunk 2]
```diff
@@ -7,9 +7,10 @@ env:
     - REPO=docksal/varnish
 
   matrix:
-    - VERSION=6.1 TAGS=6.1,6,latest
-    - VERSION=5.2 TAGS=5.2,5
-    - VERSION=4.1 TAGS=4.1,4
+    - VERSION=6.1.1 TAGS=6.1,6,latest
+    - VERSION=6.0.2 TAGS=6.0
+    - VERSION=5.2.1 TAGS=5
```
Comment (lmakarov) [inline .travis.yml]: Drop 5.2.1 since it's no longer supported (retired), according to
https://varnish-cache.org/releases/

File: Dockerfile [hunk 3]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
 ENV VARNISH_VARNISHNCSA_PARAMS ''
 ENV VARNISH_CACHE_TAGS_HEADER Cache-Tags
 
+COPY patches /tmp/patches/
+COPY config /tmp/config/
+COPY scripts /tmp/scripts/
+COPY healthcheck.sh /usr/bin/healthcheck.sh
+COPY startup.sh /usr/bin/startup.sh
```
Comment (lmakarov) [inline Dockerfile]: Let's standardize on using `/usr/local/bin/docker-entrypoint.sh` for the main entrypoint script.
Renamed the source file as well.

File: Dockerfile [hunk 4]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
 ENV VARNISH_VARNISHNCSA_PARAMS ''
 ENV VARNISH_CACHE_TAGS_HEADER Cache-Tags
 
+COPY patches /tmp/patches/
+COPY config /tmp/config/
+COPY scripts /tmp/scripts/
+COPY healthcheck.sh /usr/bin/healthcheck.sh
```
Comment (lmakarov) [inline Dockerfile]: Let's standardize on using `/usr/local/bin/docker-healthcheck.sh` for the healthcheck script.

File: Dockerfile [hunk 5]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
...
+COPY startup.sh /usr/bin/startup.sh
+
+RUN set -ex && apk add --update --no-cache bash && /tmp/scripts/build
```
Comment (lmakarov) [inline Dockerfile]: Please use the same RUN format as in other images, e.g.
https://github.com/docksal/service-vhost-proxy/blob/develop/Dockerfile#L45

File: Dockerfile [hunk 6]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
...
+
 EXPOSE 80
 EXPOSE 6082
```
Comment (lmakarov) [inline Dockerfile:28]: Add `ENTRYPOINT` definition

File: README.md [hunk 7]
```diff
@@ -1,7 +1,9 @@
 # Varnish Docker images for Docksal
 
-- Varnish 4.1
+- Varnish 6.1
+- Varnish 6.0
 - Varnish 5.2
```
Comment (lmakarov) [inline README.md:5]: Drop 5.2

File: config/6.1/default.vcl [hunk 8]
```diff
@@ -0,0 +1,192 @@
+# Based on https://fourkitchens.atlassian.net/wiki/display/TECH/Configure+Varnish+3+for+Drupal+7
```
Comment (lmakarov) [inline config/6.1/default.vcl]: Drop commends on lines 1-18, as they are no relevant (do this for all versions)

File: config/default/default.vcl [hunk 9]
```diff
@@ -0,0 +1,192 @@
+# Based on https://fourkitchens.atlassian.net/wiki/display/TECH/Configure+Varnish+3+for+Drupal+7
```
Comment (lmakarov) [inline config/default/default.vcl]: Why do we need this, if thee are configs per version anyway? If individual version configs are identical, then we should use a single one instead.

File: Dockerfile [hunk 10]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
 ENV VARNISH_VARNISHNCSA_PARAMS ''
 ENV VARNISH_CACHE_TAGS_HEADER Cache-Tags
 
+COPY patches /tmp/patches/
```
Comment (lmakarov) [inline Dockerfile:16]: Use `/opt/patches` instead

File: Dockerfile [hunk 11]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
...
+COPY config /tmp/config/
```
Comment (lmakarov) [inline Dockerfile:17]: Use `/opt/config` instead

File: Dockerfile [hunk 12]
```diff
@@ -22,10 +13,18 @@ ENV VARNISH_VARNISHD_PARAMS ''
...
+COPY scripts /tmp/scripts/
```
Comment (lmakarov) [inline Dockerfile:18]: Use `/opt/scripts` instead

**Review by lmakarov — APPROVED** [review]

File: Dockerfile (hunk 12 above)
Comment (sergey-zabolotny) [reply Dockerfile:18]: all files placed in /tmp are only needed during the build. after the build finished, everything in the /tmp directory is deleted. I do not see any reason to put the files necessary only for build process in any place other than the tmp directory

**Comment by lmakarov** [comment]
Replaced by #35

**Pull request closed without merging**